### DISCUSSION TITLE
RAW and JPEG Sidecar

### DISCUSSION THREAD

**Discussion by farnots** [photoprism/photoprism#2057] [Q&A]
Hello, 

I'm shooting with my Camera in RAW + JPEG. According to this [discussion](https://github.com/photoprism/photoprism/discussions/1684), Prims shouldn't create a JPEG thumbnail from my RAW file but use directly the sidecar JPEG file.

In my case I've multiple JPEG file : 
- P2130078.jpg : The one from my camera
- P2130078.JPG : The one generated by Prism
- P2130078.ORF : My RAW original photos
- P2130078.xmp : The XMP file generated by Lightroom

![CleanShot 2022-02-21 at 16 34 51](https://user-images.githubusercontent.com/3254286/154986316-3b4bd8f2-3b49-4559-a4ae-40f8c27a9e53.png)

Is it the case of the file that cause issue ? Also the primary photos uses by Prism is 

Also, what appends if I generated a new JPEG photo from Lightroom. For exemple I made edit and export to `P2130078.jpg`. Does PhotoPrism will update the current photos and remove the thumbnail ? 

----

Side question, I've some ORF (Olympus Raw files) in which I updated the rotation of the photos. Lightroom saves my changes to the XMP files but the generated file from Prism doesn't take this information into account : 

Exemple (Edit exported from Lightroom) : 
![CleanShot 2022-02-21 at 16 45 11](https://user-images.githubusercontent.com/3254286/154987490-b499457a-a1ff-4975-856b-37a9546685b1.png)

Thumbnail generated from Prism based on the RAW : 
![CleanShot 2022-02-21 at 16 45 56](https://user-images.githubusercontent.com/3254286/154987575-5429c598-c2a5-4562-8afe-709563c16897.png)

**Comment by lastzero** [comment]
Was the manually created JPEG already present or did you add it later? The automatically generated JPEG will not be deleted in this case, and you must manually change the primary JPEG in the Files tab of the Edit dialog.

JPEGs are currently not regenerated when XMP files change. If the change is only related to the metadata and not to the appearance of the image, this would cause a huge overhead. Also, the rendering information in your XMP may not be compatible with Darktable, for example, if you use Photoshop.

**Reply by lastzero** [reply]
We have updated our FAQs as this question has been asked a few times now:
https://docs.photoprism.app/getting-started/faq/#are-jpegs-updated-when-raw-or-xmp-files-change

**Reply by farnots** [reply]
The JPEG was already present since it came from my camera. I generated and overwrite the current one when exporting my edit. In don't understand why the conversion are made. 

Should I avoid RAW conversion to avoid this case ? 

Ok maybe is the Lightroom XMP that is not compatible. Thanks. But if I update my sidecar photo, the changes will be available in prism on next indexation ?

**Reply by lastzero** [reply]
Did you test with other images to make sure that the issue is not caused by broken or unreadable files only in this case?

If there is a JPEG, a new one should not be created.... when you provide us with test files, we have the possibility to debug the problem. Otherwise, it's a lot of (unhelpful) guesswork.

We recommend manually updating existing JPEG sidecar files as needed or creating additional JPEGs, so you can choose between different versions. New files and other metadata changes are detected and reflected in the index as usual when your library is scanned.

**Reply by lastzero** [reply]
Did you run the convert command manually? I'm not sure, but that could make a difference. The upper/lower case file extensions should also be investigated. The behavior depends on your file system(s), which we have no information about.

**Reply by lowrents** [reply]
Hey! 

this weekend I was able to reach my long term goal: have a proper pipeline from the SD card of the camera to a photo management software, hosted on a local server to avoid using iCloud Photos - and PhotoPrism was the final pice of the puzzle to achieve this! 

So first of all a big thank you for this great project! I really appreciate your work. 

I also had the same situation where I want to backup RAWs from my camera into my file system on the NAS (reference PhotoPrism library). I tried to regenerate the thumbnails and re-index the originals in hope to get rid of orphaned RAF.jpgs but I realised that timing and presence of the files are key. 

The problem with the sidecars got even worst, when I processed the RAW with exiftool and change the extensions case (.RAF -> .raf) which means that I have 4 images stack in some situation. (.RAF.jpg, .raf.jpg .jpg, .raf). 
So I did some manual clean up, also with the .heic images by converting the original in .jpg to avoid generating sidecar files. After I did some testing on single images and saw that after a rescan the application runs without issues and realised that the files did't exist. I was wondering if there is still a record anywhere because the log says: 


basically I was in the location  `/application/photoprism` and ran some 
`find . -name "*.heic.jpg" -exec rm -rf {} \;`
commands. 
using `du -hc --max-depth=1` to track my progress I reduces my sidecar storage from roughly 46 GB to 1.2GB

I am right that the [PhotoPrism CLI](https://docs.photoprism.app/getting-started/docker-compose/#command-line-interface) didn't cover this? I thought it would be a nice idea to have some feature which lets you recreate and/or tidy up the `/application/photoprism/storage/sidecar`. Because I also realised that the RAF files which doesn't have a related JPG didn't get touched a second time by the RAW conversion.

Sorry in case this is mentioned anywhere and I missed it, because I found a [docu page](https://docs.photoprism.app/user-guide/advanced/commands/) but no other related posts. 

Thanks for your thoughts. 

Cheers!

**Reply by graciousgrey** [reply]
The page in the docs you mention is not finished and should be removed. 
Just  use  `docker-compose exec photoprism photoprism help` instead to see a list of all commands with  a description.

**Reply by lowrents** [reply]
Thanks for the hint. I was just thinking that this topic/ feature might be considered elsewhere but got postponed or canceled for some reasons even because the page in the docs seems not to be finished.
So it was not planned to implement some kind of sidecar cleanup?

**Reply by graciousgrey** [reply]
You can find cleanup and purge commands in the command help ;)

**Reply by lowrents** [reply]
Okay wow... I am so dump and answered without even testing :D Sorry for asking the obvious twice!

I thought that the `docker-compose exec photoprism photoprism help` command would return the same list as in the [CLI Examples](https://docs.photoprism.app/getting-started/docker-compose/#examples). But now I realised that these are only example and the actual command set is different.

I couldn't access my server running PhotoPrism yesterday that's why I couldn't execute the command and test it by my own (and this is also the reason for my late response, sorry for that).

Thanks again for your help and this awesome software.
